src/test/java/com/cag/cagbackendapi/integration/ProfileControllerIntegrationTests.kt

```diff
@@ -186,4 +195,121 @@ class ProfileControllerIntegrationTests {
+        assertNotNull(createUser.userId)
+
+        //get the created profile
+        assertNotNull(getProfileResponse)
+        assertEquals(HttpStatus.OK, getProfileResponse.statusCode)
+        assertNotNull(getProfile.profile_id)
+        assertEquals(userProfile.pronouns, getProfile.pronouns)
+        assertEquals(userProfile.lgbtqplus_member, getProfile.lgbtqplus_member)
+        assertEquals(userProfile.gender_identity, getProfile.gender_identity)
+        assertEquals(userProfile.comfortable_playing_man, getProfile.comfortable_playing_man)
+        assertEquals(userProfile.comfortable_playing_women, getProfile.comfortable_playing_women)
+        assertEquals(userProfile.comfortable_playing_neither, getProfile.comfortable_playing_neither)
+        assertEquals(userProfile.comfortable_playing_transition, getProfile.comfortable_playing_transition)
+        assertEquals(userProfile.height_inches, getProfile.height_inches)
+        assertEquals(userProfile.agency, getProfile.agency)
+        assertEquals(userProfile.website_link_one, getProfile.website_link_one)
+        assertEquals(userProfile.website_link_two, getProfile.website_link_two)
+        assertEquals(userProfile.website_type_one, getProfile.website_type_one)
+        assertEquals(userProfile.website_type_two, getProfile.website_type_two)
+        assertEquals(userProfile.bio, getProfile.bio)
+    }
+
+    @Test
+    fun getProfile_invalidUserId_400BadRequest() {
+        val userId = null
+
+        //get profile headers
+        val headers3 = HttpHeaders()
+        headers3.set("authKey", validAuthKey)
+        headers3.set("userId", userId)
+        val request3 = HttpEntity(null, headers3)
+
+        //create error message
+        val errorDetailsResponse = testRestTemplate.exchange("/user/$userId/profile", HttpMethod.GET, request3, ErrorDetails::class.java)
+
+        //check the error
+        assertEquals(HttpStatus.BAD_REQUEST, errorDetailsResponse.statusCode)
+        assertNotNull(errorDetailsResponse?.body?.time)
+        assertEquals(errorDetailsResponse?.body?.restErrorMessage, RestErrorMessages.BAD_REQUEST_MESSAGE)
+        assertEquals(errorDetailsResponse?.body?.detailedMessage, DetailedErrorMessages.INVALID_USER_ID)
+    }
+
+    @Test
+    fun getProfile_noProfileForUser_404NotFound() {
+        //create user headers
+        val headers = HttpHeaders()
+        headers.set("authKey", validAuthKey)
+        val request = HttpEntity(validRegisterUser, headers)
+
+        //create user
+        val createdUserResponse = testRestTemplate.postForEntity("/user/register", request, String::class.java)
+        val createUser = objectMapper.readValue(createdUserResponse.body, UserDto::class.java)
+        val userIdUUID = createUser.userId
+
+
+        //get profile headers
+        val headers3 = HttpHeaders()
+        headers3.set("authKey", validAuthKey)
+        headers3.set("userId", userIdUUID.toString())
```

Comment:
The UserId is sent in the path, not in the header! You also have it in the path so this test will pass, but this header is just extra unneeded data